File: app-miyou/src/main/java/com/miyou/app/infrastructure/monitoring/repository/SpringDataPerformanceMetricsRepository.java
package com.miyou.app.infrastructure.monitoring.repository;

import java.time.Instant;

import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.monitoring.document.PerformanceMetricsDocument;
import reactor.core.publisher.Flux;

public interface SpringDataPerformanceMetricsRepository
	extends
		ReactiveMongoRepository<PerformanceMetricsDocument, String> {

	Flux<PerformanceMetricsDocument> findByStartedAtBetweenOrderByStartedAtDesc(Instant startTime,
		Instant endTime);

	Flux<PerformanceMetricsDocument> findByStatusOrderByStartedAtDesc(String status,
		Pageable pageable);

	Flux<PerformanceMetricsDocument> findByTotalDurationMillisGreaterThanEqual(long thresholdMillis,
		Pageable pageable);

	Flux<PerformanceMetricsDocument> findAllByOrderByStartedAtDesc(Pageable pageable);
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/monitoring/repository/SpringDataPerformanceMetricsRepository.kt
package com.miyou.app.infrastructure.monitoring.repository

import com.miyou.app.infrastructure.monitoring.document.PerformanceMetricsDocument
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux
import java.time.Instant

interface SpringDataPerformanceMetricsRepository : ReactiveMongoRepository<PerformanceMetricsDocument, String> {
    fun findByStartedAtBetweenOrderByStartedAtDesc(
        startTime: Instant,
        endTime: Instant,
    ): Flux<PerformanceMetricsDocument>

    fun findByStatusOrderByStartedAtDesc(
        status: String,
        pageable: Pageable,
    ): Flux<PerformanceMetricsDocument>

    fun findByTotalDurationMillisGreaterThanEqual(
        thresholdMillis: Long,
        pageable: Pageable,
    ): Flux<PerformanceMetricsDocument>

    fun findAllByOrderByStartedAtDesc(pageable: Pageable): Flux<PerformanceMetricsDocument>
}
